build.gradle

```diff
@@ -1,104 +1,124 @@
-}
-
-dependencies {
-    compileOnly 'javax.servlet:javax.servlet-api:3.0.1'
-    compileOnly 'javax.servlet.jsp:javax.servlet.jsp-api:2.2.1'
-    compileOnly 'javax.servlet.jsp.jstl:javax.servlet.jsp.jstl-api:1.2.1'
-
-    testCompile 'junit:junit:4.12'
-    testCompile 'org.springframework:spring-test:4.1.7.RELEASE'
-}
-
-jar {
-    manifest {
-        attributes 'Implementation-Title': 'Java 8 java.time JSP tags',
-                   'Implementation-Version': version
-    }
-}
-
-task javadocJar(type: Jar) {
-    classifier = 'javadoc'
-    from javadoc
-}
-
-task sourcesJar(type: Jar) {
-    classifier = 'sources'
-    from sourceSets.main.allSource
-}
-
-artifacts {
-    archives javadocJar, sourcesJar
-}
-
-install {
-    repositories.mavenInstaller {
-        pom.project {
-            name 'Java 8 java.time JSP tags'
-            description 'JSP tag support for Java 8 java.time (JSR-310)'
-            url 'https://github.com/sargue/java-time-jsptags'
-
-            scm {
-                connection 'scm:git:git@github.com:sargue/java-time-jsptags.git'
-                developerConnection 'scm:git:git@github.com:sargue/java-time-jsptags.git'
-                url 'git@github.com:sargue/java-time-jsptags.git'
-            }
-
-            licenses {
-                license {
-                    name 'The Apache License, Version 2.0'
-                    url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
-                }
-            }
-
-            developers {
-                developer {
-                    id 'sargue'
-                    name 'Sergi Baila'
-                    email 'sargue@gmail.com'
-                }
-            }
-        }
-    }
-}
-
-bintray {
-    user = project.hasProperty('BINTRAY_USER') ? BINTRAY_USER : ''
-    key = project.hasProperty('BINTRAY_KEY') ? BINTRAY_KEY : ''
-    configurations = ['archives']
-    pkg {
-        repo = 'maven'
-        name = 'net.sargue:java-time-jsptags'
-        licenses = ['Apache-2.0']
-        vcsUrl = 'https://github.com/sargue/java-time-jsptags'
-        version {
-            name = project.version
-            desc = 'JSP tag support for Java 8 java.time (JSR-310)'
-
-            gpg {
-                sign = true
-                passphrase = project.hasProperty('BINTRAY_GPG') ? BINTRAY_GPG : ''
-            }
-        }
-    }
-}
+plugins {
+    id "com.jfrog.bintray" version "1.7.3"
+    id "java"
+    id "eclipse"
+    id "com.github.ben-manes.versions" version "0.41.0" // adds dependencyUpdates task
+    id "maven-publish"
+}
+
+group = 'net.sargue'
+archivesBaseName = 'java-time-jsptags'
+version = '2.0.0'
+
+compileJava.options.encoding = 'UTF-8'
+compileTestJava.options.encoding = 'UTF-8'
+
+java {
+    // set version of Java that the source confirms to. 
+    // The bytecode will be for this version of Java as well, unless targetCompatibility is specified.
+    sourceCompatibility = JavaVersion.VERSION_17
+}
+
+repositories {
+    mavenLocal()
```

Comment:
That's the standard setup I use and I needed to publish it locally to use it elsewhere.

Reply:
I would prefer not to add this, it can be a [source for inconsistencies](https://docs.gradle.org/current/userguide/declaring_repositories.html#sub:local-repos) and it's [not recommended](https://docs.gradle.org/current/userguide/declaring_repositories.html#sec:case-for-maven-local).

There are [better ways]( https://medium.com/decisionbrain/when-to-not-use-mavenlocal-in-your-gradle-build-script-6da03902f9df) to handle local dependencies with gradle.

Reply:
Thanks for the pointers. It's been removed from this PR and I'll be looking at my other projects.

Reply:
Are you sure you've removed the `mavenLocal()` repository? I still see it.

Reply:
Sorry, I thought I had. It should be gone now.